pennylane/devices/qubit/preprocess.py

```diff
@@ -96,61 +99,140 @@ def _operator_decomposition_gen(
 #######################
 
 
-def validate_multiprocessing_workers(max_workers):
+@transform
+def validate_device_wires(
+    tape: qml.tape.QuantumTape, device
```

Comment:
bit of a nitpick but maybe we can just pass the wires? The error won't say the device name anymore but that's fine imo

Reply:
I will keep it as it is because I just want to update the device for the transform program, but we can come back to it in an upcoming separated PR.